### ISSUE TITLE
AreaChart with legend and XAxis label. I am allowing the user to turn the legend on and off but by doing so the XAxis label (not ticks) goes away.

### ISSUE THREAD

**Issue by kenperregaux** [recharts/recharts#3755]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/simple-area-chart-forked-97rthz)

### Steps to reproduce
Added a legend, an XAxis lable and a boolean flag to hide the legend. The XAxis label disappears with the legend when changing the boolean flag to false. It appears to be in the Dom but no matter what I try it will not show.

### What is expected?
XAxis label still shows even thought the Legend is hidden.

### What is actually happening?
XAxis is hidden.

| Environment | Info |
|---|---|
| Recharts | v2.5.0 |
| React | 18.2 |
| System | Windows 11 Pro |
| Browser | Chrome Version 112.0.5615.138 |

**Comment by ckifer** [comment]
@kenperregaux looks like its there its just under the chart. You'll need to add some margin to the bottom of the chart so it is visible.

We should account for this and modify positioning but for now your workaround will be to keep a variable for bottom margin and change it when the legend is hidden
https://codesandbox.io/s/simple-area-chart-forked-mxdzp2?file=/src/App.tsx

**Comment by kenperregaux** [comment]
@ckifer thanks for your answer. However, margin bottom has no affect and using offset to separate it from the tick labels actually puts outside the chart.
![image](https://github.com/recharts/recharts/assets/24940607/036f6b83-9c39-499d-8520-99cedea9ec47)

**Comment by nikolasrieble** [comment]
To me, this cases seems to indicate that we are missing default margins. If some components are only visible when margins are set in a certain manner, I believe those margins should be set by default. I understand how this is a breaking change, but ihmo it is a bug fix. 

wdyt @ckifer

**Comment by ckifer** [comment]
Yeah I mean it's a bug/enhancement - the only way it's a breaking change is if a user currently sets dynamic margin. Adjusting it in the library will give them a poor result, but the remediation would be to remove that hack. 

It's a bug fix/enhancement - as of right now auto adjusting margin internally isn't accounted for in legend, as it is a feature that doesn't exist I would consider it an enhancement. On the same side everything should be by default visible if someone specifies it (no matter the margin that they set on the chart) hence the "bug" if you could call it that. 

Either way I think we could do this and release in a minor without any issue

**Comment by ckifer** [comment]
related https://github.com/recharts/recharts/issues/3590
